pulpcore/tests/functional/__init__.py

```diff
@@ -812,9 +812,18 @@ def _monitor_task(task_href):
 @pytest.fixture(scope="session")
 def monitor_task_group(task_groups_api_client):
     def _monitor_task_group(task_group_href):
+        def tasks_running(tg):
+            return (tg.waiting + tg.running) > 0
+
+        def group_wait(tg):
+            return not tg.all_tasks_dispatched or tasks_running(tg)
+
+        def group_fail(tg):
+            return not tg.all_tasks_dispatched and tg.failed > 0 and not tasks_running(tg)
```

Comment:
```suggestion
            return tg.failed > 0 and not tasks_running(tg)
```
Would this also be a fail condition?

Reply:
Still, I do not see any reason for checking the presence of running tasks. If some tasks fail, we can consider the group failed regardless of the running tasks.

Reply:
As long as a single sub-task is running, the taskgroup is active. We may already know something did not succeed, but what the tests really care about is that the taskgroup is in a final state.
Maybe we should adjust the names of the functions to better reflect this.